platforms/ide/problems-api/src/integTest/groovy/org/gradle/api/problems/ProblemsServiceIntegrationTest.groovy

```diff
@@ -58,6 +58,51 @@ class ProblemsServiceIntegrationTest extends AbstractIntegrationSpec {
         }
     }
 
+    def "correct location for space-assignment deprecation"() {
+        buildFile '''
+            class GroovyTask extends DefaultTask {
+                @Input
+                def String prop
+                void doStuff(Action<Task> action) { action.execute(this) }
+            }
+            tasks.withType(GroovyTask) { conventionMapping.prop = { '[default]' } }
+            task test(type: GroovyTask)
+            assert test.prop == '[default]'
+            test {
+                description 'does something'
+            }
+            assert test.description == 'does something'
+            test.doStuff {
+                prop = 'new value'
+            }
+            assert test.prop == 'new value'
+'''
+
+        executer.expectDocumentedDeprecationWarning(
+            "Space-assignment syntax in Groovy DSL has been deprecated. " +
```

Comment:
I will not block the PR on this, but I'd like to point out that preserving this test will be very hard when space-assignment syntax will actually become a problem. Maybe we should leave a note on the test with the following points:
- The root cause AsmBackedClassGenerator injecting a deprecation that result in invalid location info
- Space-assignment syntax is removed then the test could either be deleted, or updated if AsmBackedClassGenerator still contains a deprecation invocation

Reply:
I actually think we should come up with a test case that works without the deprecation.
If somebody deletes it in 10 we'll lose coverage.

Reply:
Removing the deprecated features will make this test fail. With the comments you added, the next person should be able to figure out what needs to be done.